src/main.rs

```diff
@@ -309,25 +373,35 @@ impl App {
             None
         }
     }
+    fn hover(&self, params: TextDocumentPositionParams) -> Option<(Option<Range>, String)> {
+        let (_, content, tree) = self.files.get(&params.text_document.uri)?;
+        let offset = utils::lookup_pos(content, params.position)?;
+        let child_tree = climb_tree(tree.as_ref().ok()?, offset).clone();
+        let range = utils::range(content, child_tree.range?);
+        let val = child_tree.eval().ok()?.format_markdown();
```

Comment:
Wouldn't it make sense here to give a response with an eval-error? (or am I misundestanding this?)

Reply:
Maybe I should separate the error enum into a value vs internal errors?

1. Value errors, such as division by zero or type errors. The language server would display these issues
2. Internal evaluator errors, such as parsing issues or unimplemented features. Since this PR is fairly minimal, it returns an error for anything other than arithmetic. I assume we'd want to silently handle this for now

Reply:
Yup, makes sense to me :+1:

Reply:
Perfect. I've pushed some code that addresses error handling